force-app/main/default/aura/CompensationTable/CompensationTableHelper.js

```diff
@@ -0,0 +1,32 @@
+({
+	getCompensations : function (component) {
+		var action = component.get("c.getCompensations");
+        action.setCallback(this,function(response){
+            if (response.getState() === "SUCCESS"){
+                component.set("v.compensationDetails",response.getReturnValue());
+            }
+            else
+            {
+                alert('Error retrieving compensation details.');
+            }
+        });
+        $A.enqueueAction(action);
+	},
+    filter : function (component){
+        var filter = component.find("select").get("v.value");
+        var rows = document.getElementById("table").rows;
+        for(var i=1; i<rows.length; i++){
```

Comment:
please, be careful with the spaces between the values and the comparison operators. ie: 
instead of "i=1" , 
          try: "i = 1"  --> the spaces =)